client.go

```diff
@@ -180,7 +180,27 @@ func runStreamedCommand(img, cmd string, getError bool) error {
 // It returns the ID of the container spawned from the image.
 // Note well: the container is NOT deleted when the given command terminates.
 func runCommandInContainer(img string, cmd []string, dst io.Writer) (string, error) {
-	return startContainer(img, cmd, true, dst)
+	id, err := createContainer(img, cmd)
+	if err != nil {
+		log.Println(err)
+		return "", err
+	}
+
+	run_cmd:
+		id, err = startContainer(id, cmd, true, dst)
+		return id, err
+		switch err.(type) {
+		case dockerError:
+			de := err.(dockerError)
+			if isZypperExitCodeSevere(de.exitCode) {
+				if de.exitCode == zypperExitInfRestartNeeded  {
+					goto run_cmd
```

Comment:
I'm sure you can avoid using a `goto` statement 😉

Reply:
I prefer using a goto here. Golang doesn't support do-while loops, so an alternative to it would be using a for loop and variable indicating another run.